### DISCUSSION TITLE
Questions about `binary_sv2` crate

### DISCUSSION THREAD

**Discussion by jbesraa** [stratum-mining/stratum#1161] [General]
this discussion should guide the documentation of `binary_sv2` crate, as described on https://github.com/stratum-mining/stratum/issues/845

**Comment by jbesraa** [comment]
High Level Questions:

- Is there a documentation or spec the implementation adhere to?
- What is the benifit of  `binary-sv2/binary-sv2` beside rexporting `serde` and `non-serde`?
- Within `binary-sv2/no-serde-sv2` there are two crates, `codec` and `derive_codec`. Is
  `derive_codec` expected to be used by the user or by `codec` only?

**Reply by Fi3** [reply]
1. sv2 specification chapter 3
2. it only reexport them, is nice to have so you can code your implementation importing binary-sv2 and then just with a feature you chose if you want to use serde (for example you need translation from json) or not (faster and smaller)
3. this cause you need to have a separate crate if you want to define macros

**Reply by jbesraa** [reply]
> 1. sv2 specification chapter 3
> 
> 2. it only reexport them, is nice to have so you can code your implementation importing binary-sv2 and then just with a feature you chose if you want to use serde (for example you need translation from json) or not (faster and smaller)
> 
> 3. this cause you need to have a separate crate if you want to define macros  

Thank you
